src/measure_theory/measure/measure_space.lean

```diff
@@ -1824,6 +1824,73 @@ begin
     (lt_of_le_of_lt (le_iff'.1 h _) (C.finite i)), C.spanning⟩⟩,
 end
 
+section disjointed
+
+variable [measurable_space α]
+
+/-- Given `S : μ.finite_spanning_sets_in {s | measurable_set s}`,
+`disjointed_finite_spanning_sets_in` provides a `finite_spanning_sets_in {s | measurable_set s}`
+such that its underlying sets are pairwise disjoint. -/
+def disjointed_finite_spanning_sets_in (μ : measure α)
+  (S : μ.finite_spanning_sets_in {s | measurable_set s}) :
+   μ.finite_spanning_sets_in {s | measurable_set s} :=
+⟨disjointed S.set, measurable_set.disjointed S.set_mem,
+  λ n, lt_of_le_of_lt (measure_mono (disjointed_subset S.set n)) (S.finite _),
+  S.spanning ▸ Union_disjointed⟩
+
+lemma disjoint_disjointed_finite_spanning_sets_in {μ : measure α}
+  (S : μ.finite_spanning_sets_in {s | measurable_set s}) :
+  pairwise (disjoint on (disjointed_finite_spanning_sets_in μ S).set) :=
+disjoint_disjointed _
+
+/-- Given two σ-finite measures `μ` and `ν`, `finite_spanning_sets_in_add μ ν` is a sequence of
+pairwise disjoint, finite spanning sets in `{s | measurable_set s}` with respect to `μ + ν`. -/
+def finite_spanning_sets_in_add (μ ν : measure α) [sigma_finite μ] [sigma_finite ν] :
+  (μ + ν).finite_spanning_sets_in {s | measurable_set s} :=
+(μ + ν).disjointed_finite_spanning_sets_in $
+let S := μ.to_finite_spanning_sets_in in
+let T := ν.to_finite_spanning_sets_in in
+{ set := λ n : ℕ, S.set n.unpair.1 ∩ T.set n.unpair.2,
+  set_mem := λ n, (S.set_mem n.unpair.1).inter (T.set_mem n.unpair.2),
+  finite :=
+    begin
+      intro n,
+      rw [add_apply, ennreal.add_lt_top],
+      exact ⟨lt_of_le_of_lt (measure_mono (set.inter_subset_left _ _)) (S.finite n.unpair.1),
+             lt_of_le_of_lt (measure_mono (set.inter_subset_right _ _)) (T.finite n.unpair.2)⟩,
+    end,
+  spanning := by simp_rw [set.Union_unpair (λ i j, S.set i ∩ T.set j), ← set.inter_Union,
+                          ← set.Union_inter, S.spanning, T.spanning, set.inter_univ] }
+
+lemma disjoint_finite_spanning_sets_in_add
+  {μ ν : measure α} [sigma_finite μ] [sigma_finite ν] :
+  pairwise (disjoint on (finite_spanning_sets_in_add μ ν).set) :=
+disjoint_disjointed_finite_spanning_sets_in _
+
+/-- Given measures `μ`, `ν` where `ν ≤ μ`, `finite_spanning_sets_in_of_le` provides the induced
+`finite_spanning_set` with respect to `ν` from a `finite_spanning_set` with respect to `μ`. -/
+def finite_spanning_sets_in_of_le (μ ν : measure α) (h : ν ≤ μ) {C : set (set α)}
+  (S : μ.finite_spanning_sets_in C) : ν.finite_spanning_sets_in C :=
```

Comment:
If you name it this way dot notation should work:
```suggestion
def finite_spanning_sets_in.restrict {μ ν : measure α} (h : ν ≤ μ) {C : set (set α)}
  (S : μ.finite_spanning_sets_in C) : ν.finite_spanning_sets_in C :=
```

Reply:
would it be `finite_spanning_sets_in.le` instead of `finite_spanning_sets_in.restrict`? I don't think `measure.restrict` is used here

Reply:
`mono` could also be a good name

Reply:
I think `.le` would be confusing. `.of_le` would be ok, as would `.mono`.

Reply:
It seems that `mono` is taken by monotone wrt to sets. I will rename it to `.of_le`